### PR TITLE
Plugin upgrade

### PR DESCRIPTION
**Pull request by mkleina**
Summary of changes:
- Upgraded plugin and unit tests to use plugin-lib-go
- Add information about diagnostic mode to README.md and include simple bash script in examples

Testing done:
- Manual
- Unit test

### CODE REVIEW

File: collector/collector.go [hunk 1]
```diff
@@ -31,27 +31,21 @@ import (
...
 	//pluginName namespace part
-	pluginName = "snmp"
+	PluginName = "snmp"
 
 	// version of plugin
-	version = 1
-
-	//pluginType type of plugin
-	pluginType = plugin.CollectorPluginType
+	Version = 1
```
Comment (katarzyna-z) [inline collector/collector.go]: Please increase plugin version.

**Comment by katarzyna-z** [comment]
I know that this plugin requires configuration but I've run plugin diagnostics without any configuration and I got following output:
```
./snap-plugin-collector-snmp 
panic: runtime error: index out of range

goroutine 1 [running]:
panic(0x777820, 0xc42000a130)
	/home/kujawka/.gvm/gos/go1.7/src/runtime/panic.go:500 +0x1a1
main.main()
	/home/kujawka/.gvm/pkgsets/go1.7/global/src/github.com/intelsdi-x/snap-plugin-collector-snmp/main.go:39 +0xef

```
Is it expected behavior for that kind of plugins?

File: examples/standalone.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+./build/linux/x86_64/snap-plugin-collector-snmp --config '{
```
Comment (katarzyna-z) [inline examples/standalone.sh]: This script does not work from this localisation, path is not correct, see:
```
./standalone.sh 
./standalone.sh: line 3: ./build/linux/x86_64/snap-plugin-collector-snmp: No such file or directory
```

File: README.md [hunk 3]
```diff
@@ -380,6 +380,30 @@ $ snaptel task watch <task_id>
...
+    "user_name": "user",
+    "security_level": "AuthPriv",
+    "auth_password": "password",
+    "auth_protocol": "MD5",
+    "priv_protocol": "DES",
+    "priv_password": "password"
+}'
+```
+
+If configuration is valid, plugin should output metric catalog and collected metrics to stdout.
```
Comment (katarzyna-z) [inline README.md]: Consider changing "stdout to "standard output" but it's just suggestion.

**Comment by mkleina** [comment]
@katarzyna-z Fixed :+1:

**Pull request merged**